javascript/javascript1/week3/exercises.js

```diff
@@ -0,0 +1,267 @@
+  }
+  console.log(days);
+  console.log(hours);
+  console.log(minutes);
+
+  return minutes, hours, days;
+}
+
+function calculateMinutes() {
+  if (minutes >= 60) {
+    hours += Math.floor(minutes / 60);
+    reminderOfMinutes += minutes % 60;
+    console.log(hours);
+    console.log(reminderOfMinutes);
+    return reminderOfMinutes;
+  }
+}
+
+function calculateHoursAndDays() {
+  if (hours >= 24) {
+    days += Math.floor(hours / 24);
+    reminderOfHours += hours % 24;
+    console.log(days);
+    console.log(reminderOfHours);
+    return reminderOfHours;
+  }
+}
+
+function logSumWatching() {
+  const sumWatching = `${days} days ${reminderOfHours} hours ${reminderOfMinutes} minutes`;
+  console.log(sumWatching);
+  return sumWatching;
+}
+
+function calculateWatchingGameOfThronesPercentageOfLife() {
+  const myDurationOfWatchingGameOfThronesInMinutes = (3 * 24 + 1) * 60;
+  console.log(myDurationOfWatchingGameOfThronesInMinutes);
+  watchingGameOfThronesPercentageOfLife =
+    (myDurationOfWatchingGameOfThronesInMinutes * 100) /
+    averageLifespanInMinutes;
+  console.log(
+    `Game of thrones took ${
+      Math.round(watchingGameOfThronesPercentageOfLife * 10000) / 10000
+    }% of my life`
+  );
+  return watchingGameOfThronesPercentageOfLife;
+}
+function calculateWatchingOrvillePercentageOfLife() {
+  const myDurationOfWatchingOrvilleInMinutes = (0 * 24 + 18) * 60 + 54;
+  console.log(myDurationOfWatchingOrvilleInMinutes);
+  watchingOrvillePercentageOfLife =
+    (myDurationOfWatchingOrvilleInMinutes * 100) / averageLifespanInMinutes;
+  console.log(
+    `The Orville took ${
+      Math.round(watchingOrvillePercentageOfLife * 10000) / 10000
+    }% of my life`
+  );
+  return watchingOrvillePercentageOfLife;
+}
+
+function calculateWatchingSquidGamePercentageOfLife() {
+  const myDurationOfWatchingSquidGameInMinutes = (0 * 24 + 8) * 60 + 6;
+  console.log(myDurationOfWatchingSquidGameInMinutes);
+  watchingSquidGamePercentageOfLife =
+    (myDurationOfWatchingSquidGameInMinutes * 100) / averageLifespanInMinutes;
+  console.log(
+    `Squid game took ${
+      Math.round(watchingSquidGamePercentageOfLife * 10000) / 10000
+    }% of my life`
+  );
+  return watchingSquidGamePercentageOfLife;
+}
+
+function calculateWatchingSeriesPercentageOfLife() {
+  const myDurationOfWatchingTvSeriesInPercentageOfLife =
+    watchingGameOfThronesPercentageOfLife +
+    watchingOrvillePercentageOfLife +
+    watchingSquidGamePercentageOfLife;
+  console.log(
+    `In total that is ${
+      Math.round(myDurationOfWatchingTvSeriesInPercentageOfLife * 10000) / 10000
+    }% of my life`
+  );
+  return myDurationOfWatchingTvSeriesInPercentageOfLife;
+}
+
+sumSeriesDurations();
+calculateMinutes();
+calculateHoursAndDays();
+logSumWatching();
+calculateWatchingGameOfThronesPercentageOfLife();
+calculateWatchingOrvillePercentageOfLife();
+calculateWatchingSquidGamePercentageOfLife();
+calculateWatchingSeriesPercentageOfLife();
+
```

Comment:
this works just fine, but 1) you are console.loging a bit too much. 2) you have functions for each of the series, but it would be more efficient (and it would be easier to debug) if you had one function that calculates the duration for ANY series.